crates/astria-core/src/primitive/v1/asset.rs

```diff
@@ -59,61 +69,163 @@ impl Denom {
     }
 
     #[must_use]
-    pub fn prefix(&self) -> &str {
-        &self.prefix
+    pub fn channel(&self) -> Option<&str> {
+        if self.prefix_segments.len() < 2 {
```

Comment:
I am not sure if this is always true. But I went with how this was set in `astria-bridge-withdrawer`

Reply:
this should be the second prefix segment actually (the channel closest to the start of the denom trace)